android/java/org/chromium/chrome/browser/set_default_browser/BraveSetDefaultBrowserUtils.java

```diff
@@ -98,19 +100,25 @@ public static void showBraveSetDefaultBrowserDialog(
         if (!isBottomSheetVisible) {
             isBottomSheetVisible = true;
 
+            try {
+                SetDefaultBrowserBottomSheetFragment bottomSheetDialog =
+                        SetDefaultBrowserBottomSheetFragment.newInstance(isFromMenu);
+
+                bottomSheetDialog.show(activity.getSupportFragmentManager(),
+                        "SetDefaultBrowserBottomSheetFragment");
```

Comment:
We should move below code (write to shared preferences) after try catch to inside try after showing bottomsheet dialog as we should only count once it's visible to user.

Reply:
It seems that after the exception we will not update the shared pref, because catch block has `return`. Or what did you mean?

Reply:
yes, you are right. My bad, I haven't noticed return statement.